commit edd95926fa08f0f2edcd7769288ae53932a4978b (refs/changes/90/4890/3)
Author: Brian Bennett <brian.bennett@joyent.com>
Date:   2018-09-27T14:14:16-07:00 (1 year ago)
    
    TRITON-798 CNS needs to support two acme-challenges
    Reviewed by: Alex Wilson <alex.wilson@joyent.com>
    Approved by: Alex Wilson <alex.wilson@joyent.com>

diff --git a/lib/vm-to-zones.js b/lib/vm-to-zones.js
index 5fe6d03..3ac9507 100644
--- a/lib/vm-to-zones.js
+++ b/lib/vm-to-zones.js
@@ -3,7 +3,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  *
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 module.exports = buildZonesFromVm;
@@ -191,9 +191,11 @@ function addInstance(zones, vm, ent, config) {
 			if (!zones[ent.zone][name])
 				zones[ent.zone][name] = [];
 			var recs = zones[ent.zone][name];
-			recs.push({
-				constructor: 'TXT',
-				args: [chal]
+			chal.split(' ').forEach(function (c) {
+				recs.push({
+					constructor: 'TXT',
+					args: [c]
+				});
 			});
 		}
 	}
@@ -308,6 +310,7 @@ function addService(zones, vm, ent, config) {
 			src: vm.uuid
 		});
 	}
+
 	function addACME(name) {
 		if (vm.customer_metadata === undefined)
 			return;
@@ -319,10 +322,12 @@ function addService(zones, vm, ent, config) {
 			if (!zones[ent.zone][name])
 				zones[ent.zone][name] = [];
 			var recs = zones[ent.zone][name];
-			recs.push({
-				constructor: 'TXT',
-				args: [chal],
-				src: vm.uuid
+			chal.split(' ').forEach(function (c) {
+				recs.push({
+					constructor: 'TXT',
+					args: [c],
+					src: vm.uuid
+				});
 			});
 		}
 	}
